File: tree.js
class TreeNode {
    constructor(value, leftNode=null, rightNode=null) {
        this.value = value;
        this.left = leftNode;
        this.right = rightNode;
    }
}

let treeNode1 = new TreeNode(1);
let treeNode2 = new TreeNode(10);
let root = new TreeNode(5, treeNode1, treeNode2);

function search(value, node) {
    if(!node || node.value === value) {
        return node;
    } else if(node.value < value) {
        return search(value, node.rightNode);
    } else {
        return search(value, leftNode);
    }
}

function insert(value, node) {
    if(value < node.value) {
        if(!node.leftNode) {
            node.leftNode = new TreeNode(value);
        } else {
            insert(value, leftNode);
        }
    } else if(value > node.value) {
        if(!node.rightNode) {
            node.rightNode = new TreeNode(value);
        } else {
            insert(value, node.rightNode);
        }
    }
}

File: tree.py
class TreeNode:
    def __init__(self, val, left=None, right=None):
        self.value = val
        self.leftChild = left
        self.rightChild = right     

node = TreeNode(1)
node2= TreeNode(10)
root = TreeNode(5, node, node2)        

def search(value, node):
    if node is None or node.value == value:
        return node 

    elif value < node.value:
        return search(value, node.leftChild)

    else:
        return search(value, rightNode)        


def insert(value, node):
    if value < node.value:
        if node.leftChild is None:
            node.leftChild = TreeNode(value)
        else:
            insert(value, node.leftChild)

    elif value > node.value:
        if node.rightChild is None:
            node.rightChild = TreeNode(value)
        else:
            insert(value, node.rightChild)                